mlos_core/mlos_core/optimizers/observations.py

```diff
@@ -0,0 +1,361 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""Contains classes for observations and suggestions."""
```

Comment:
I wonder if we can split this out a bit more, or there's just too many circular import issues.

```
mlos_core/data_types/
  __init__.py
  observation.py
  observations.py
  suggestion.py
```

Reply:
Also nicer for locating tests.

Reply:
I think we can, though for the conversion methods, you'd have to place the `from ... import` call inside the method and tell `pylint: disable=not-top-level-import`, in order to avoid `cyclic-import` issues.

Reply:
I tried this, and it is incredibly painful to get it to work